app/views/course/assessment/assessments/_assessment.html.slim

```diff
@@ -8,8 +8,12 @@
 
   - if current_course.gamified?
     td.table-base-exp = assessment.base_exp
-    - if show_time_bonus_exp?
-      td.table-time-bonus-exp = assessment.time_bonus_exp
+  - if show_bonus_attributes?
+    td.table-time-bonus-exp
+      - if assessment.bonus_end_at.present? && assessment.time_bonus_exp > 0
```

Comment:
okay I see, you mean the validation in lesson plan model ? 

I realised that there are some inconsistencies in the code, the EXP awarding code assumes that if theres' no `bonus_end_at`, then bonus_exp will always be awarded.